apps/passport-server/src/services/telegramService.ts

```diff
@@ -976,7 +977,9 @@ export class TelegramService {
       span?.setAttribute("validEventIds", validEventIds);
 
       const eventsByChat = await fetchEventsPerChat(this.context.dbPool);
-      const telegramChatId = findChatByEventIds(eventsByChat, validEventIds);
+      if (!eventsByChat.find((e) => e.telegramChatID === telegramChatId))
```

Comment:
Don't we still need to check validEventIds here? I think all this checks is that telegramChatId is associated with one of the events here, not necessarily that it's the event that matches this proof's valid event ids